htlcswitch/link.go

```diff
@@ -2329,6 +2372,7 @@ func (l *channelLink) CheckHtlcForward(payHash [32]byte,
 	// First check whether the outgoing htlc satisfies the channel policy.
 	err := l.canSendHtlc(
 		policy, payHash, amtToForward, outgoingTimeout, heightNow,
+		originalScid,
```

Comment:
If this is the `scid` of the _outgoing_ channel (this link), then the link already has this context.

Reply:
Since the link's short channel id may be one we don't want to leak, this returns a channel update with the scid used in the onion packet - spec requires this